src/Internal/Plutip/Server.purs

```diff
@@ -270,19 +270,39 @@ startPlutipContractEnv
        , clearLogs :: Aff Unit
        }
 startPlutipContractEnv plutipCfg distr cleanupRef = do
-  configCheck plutipCfg
-  startPlutipServer'
-  ourKey /\ response <- startPlutipCluster'
-  startOgmios' response
-  startKupo' response
-  { env, printLogs, clearLogs } <- mkContractEnv'
-  wallets <- mkWallets' env ourKey response
-  pure
-    { env
-    , wallets
-    , printLogs
-    , clearLogs
-    }
+  configChecked <- try (configCheck plutipCfg)
+  case configChecked of
+    Left err -> throwError $ error $ "Config check failed: " <> message err
+    Right _ -> do
+      serverStarted <- try startPlutipServer'
+      case serverStarted of
+        Left err -> throwError $ error $ "Could not start Plutip server: " <>
+          message err
+        Right _ -> do
+          clusterStarted <- try startPlutipCluster'
+          case clusterStarted of
+            Left err -> throwError $ error $ "Could not start Plutip cluster: "
+              <> message err
+            Right (ourKey /\ response) -> do
+              ogmiosStarted <- try (startOgmios' response)
+              case ogmiosStarted of
+                Left err -> throwError $ error $ "Could not start Ogmios: " <>
+                  message err
+                Right _ -> do
+                  kupoStarted <- try (startKupo' response)
+                  case kupoStarted of
+                    Left err -> throwError $ error $ "Could not start Kupo: " <>
+                      message err
+                    Right _ -> do
+                      { env, printLogs, clearLogs } <-
+                        mkContractEnv'
+                      wallets <- mkWallets' env ourKey response
+                      pure
```

Comment:
There's too much nesting. Create a helper and move the error message handling inside the functions.

Reply:
Was planning to do this anyway: this was a draft so that I could make sure I wasn't missing something.